src/graph/node.js

```diff
@@ -40,6 +41,13 @@ class Node {
   getPathsTo() {
     return this.edges.filter((e) => e.toNode.id === this.id);
   }
+
+  getDistance(node) {
+    if (!(node.name in this.distance)) {
```

Comment:
do not use name, use node.id for comparisons, id is guaranteed to be unique

Reply:
I have changed this.distance in node.js as the object with proxy and used node.id for comparisons.